drivers/dma/dma_mcux_edma.c

```diff
@@ -358,26 +374,93 @@ static int dma_mcux_edma_configure(const struct device *dev, uint32_t channel,
 	EDMA_EnableChannelInterrupts(DEV_BASE(dev), hw_channel, kEDMA_ErrorInterruptEnable);
 
 	if (block_config->source_gather_en || block_config->dest_scatter_en) {
-		EDMA_InstallTCDMemory(p_handle, DEV_CFG(dev)->tcdpool[channel],
-				      CONFIG_DMA_TCD_QUEUE_SIZE);
+        if (config->cyclic) {
+		/* Loop SG mode */
+		data->transfer_settings.write_idx = 0;
+		data->transfer_settings.empty_tcds = CONFIG_DMA_TCD_QUEUE_SIZE;
+
+		EDMA_PrepareTransfer(
```

Comment:
I'm unclear why we prepare a transfer here, and then configure each TCD using it. It seems like we will overwrite the `SADDR`, `DADDR`, `CITER`, and `BITER` fields later on in the loop below. Why not just initialize the TCDs directly in that loop? We could set `DLAST_SGA` directly as well in the loop below, rather than calling `EDMA_TcdSetTransferConfig`

Reply:
Just want to use this API in SDK driver instead of writing a new function.
yes, we can write a new function to do it as well.